lib/src/util/dart_type_utilities.dart

```diff
@@ -188,43 +188,63 @@ class DartTypeUtilities {
   static bool isNullLiteral(Expression expression) =>
       expression?.unParenthesized is NullLiteral;
 
-  static PropertyAccessorElement lookUpGetter(MethodDeclaration node) =>
-      (node.parent as ClassOrMixinDeclaration)
-          .declaredElement
-          .lookUpGetter(node.name.name, node.declaredElement.library);
+  static PropertyAccessorElement lookUpGetter(MethodDeclaration node) {
+    final parent = node.declaredElement.enclosingElement;
+    if (parent is ClassElement) {
+      return parent.lookUpGetter(node.name.name, node.declaredElement.library);
+    }
+    // todo (pq): consider extensions
+    return null;
+  }
 
   static PropertyAccessorElement lookUpInheritedConcreteGetter(
-          MethodDeclaration node) =>
-      (node.parent as ClassOrMixinDeclaration)
-          .declaredElement
-          .lookUpInheritedConcreteGetter(
-              node.name.name, node.declaredElement.library);
-
-  static MethodElement lookUpInheritedConcreteMethod(MethodDeclaration node) =>
-      (node.parent as ClassOrMixinDeclaration)
-          .declaredElement
-          .lookUpInheritedConcreteMethod(
-              node.name.name, node.declaredElement.library);
+      MethodDeclaration node) {
+    final parent = node.declaredElement.enclosingElement;
+    if (parent is ClassElement) {
+      return parent.lookUpInheritedConcreteGetter(
+          node.name.name, node.declaredElement.library);
+    }
+    // todo (pq): consider extensions
```

Comment:
Extensions don't inherit anything, so returning `null` seems like the right behavior here. I think you can remove the todo from the "inherited" versions of these methods.